File: CodeChef/src/FunWithRotation.java
import java.util.Scanner;

class FunWithRotation 
{
	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int m = sc.nextInt();
		int arr[] = new int[n];
		for(int f=0;f<n;f++)
		{
			arr[f] = sc.nextInt();
		}
		
		int shift=0;
		for(int e=0;e<m;e++)
		{
			String code = sc.next();
		//	char c = sc.
			int d = sc.nextInt();
			long temp[] = new long[d];
			
			if(code.charAt(0)=='R')
			{
				System.out.println(arr[(shift+d-1)%n]);
			}
			if(code.charAt(0)=='C')
			{
				shift = (shift + d)%n;
			}
			if(code.charAt(0)=='A')
			{
				shift = (shift-d)%n;
			}
		}
	}
}

File: CodeChef/src/FunWithRotation.c
#include<stdio.h>
#include<stdlib.h>

int main()
{
	int n,m;
	scanf("%d %d",&n,&m);
	int arr[n+2];
	int f;
	for(f=0;f<n;f++)
	{
		scanf("%d",&arr[f]);
	}


	int shift =0;
	int e;
	for(e=0;e<m;e++)
	{
		char code;
		int d;
		scanf(" %c %d",&code,&d);

		if(code=='R')
		{
			printf("%d\n",arr[(((shift+d-1)%n)+n)%n]);
		}

		if(code=='C')
		{
			shift = (((shift + d)%n)+n)%n;
		}

		if(code=='A')
		{
			shift = (((shift-d)%n)+n)%n;
		}
	}
	return 0;
}